packages/advanced-logic/src/extensions/payment-network/erc777/stream.ts

```diff
@@ -36,64 +36,174 @@ export default class Erc777StreamPaymentNetwork<
-    ) as ExtensionTypes.IAction<TCreationParameters>;
+    throw Error(
+      'masterRequestId, previousRequestId and recurrenceNumber must be all empty or all filled',
+    );
   }
 
+  /**
+   * Applies a creation extension action
+   *
+   * @param extensionAction action to apply
+   * @param timestamp ?
+   *
+   * @returns state of the extension created
+   */
   protected applyCreation(
     extensionAction: ExtensionTypes.IAction,
     timestamp: number,
   ): ExtensionTypes.IState {
-    if (!extensionAction.parameters.expectedStartDate) {
-      throw Error('expectedStartDate should not be empty');
-    }
-    if (!extensionAction.parameters.expectedFlowRate) {
-      throw Error('expectedFlowRate should not be empty');
-    }
+    /* Master request Creation */
     if (
-      extensionAction.parameters.expectedStartDate &&
-      !Utils.amount.isValid(extensionAction.parameters.expectedStartDate)
+      !(
+        extensionAction.parameters.masterRequestId ||
+        extensionAction.parameters.previousRequestId ||
+        extensionAction.parameters.recurrenceNumber
+      )
     ) {
-      throw Error('expectedStartDate is not a valid amount');
+      if (!extensionAction.parameters.expectedStartDate) {
+        throw Error('expectedStartDate should not be empty');
+      }
+      if (!extensionAction.parameters.expectedFlowRate) {
+        throw Error('expectedFlowRate should not be empty');
+      }
+      if (
+        extensionAction.parameters.expectedStartDate &&
+        !Utils.amount.isValid(extensionAction.parameters.expectedStartDate)
+      ) {
+        throw Error('expectedStartDate is not a valid amount');
+      }
+      if (
+        extensionAction.parameters.expectedFlowRate &&
+        !Utils.amount.isValid(extensionAction.parameters.expectedFlowRate)
+      ) {
+        throw Error('expectedFlowRate is not a valid amount');
+      }
+      const proxyPNCreationAction = super.applyCreation(extensionAction, timestamp);
+
+      return {
+        ...proxyPNCreationAction,
+        events: [
+          {
+            name: 'create',
+            parameters: {
+              expectedStartDate: extensionAction.parameters.expectedStartDate,
+              expectedFlowRate: extensionAction.parameters.expectedFlowRate,
+              paymentAddress: extensionAction.parameters.paymentAddress,
+              refundAddress: extensionAction.parameters.refundAddress,
+              salt: extensionAction.parameters.salt,
+            },
+            timestamp,
+          },
+        ],
+        values: {
+          ...proxyPNCreationAction.values,
+          expectedStartDate: extensionAction.parameters.expectedStartDate,
+          expectedFlowRate: extensionAction.parameters.expectedFlowRate,
+          previousRequestId: extensionAction.parameters.previousRequestId,
+          masterRequestId: extensionAction.parameters.masterRequestId,
+          recurrenceNumber: extensionAction.parameters.recurrenceNumber,
```

Comment:
Because of the condition on line 108, all of the bellow will be null or undefined, this might make it more obvious. Not sure about referenceNumber for master request though
```suggestion
          previousRequestId: null,
          masterRequestId: null,
          recurrenceNumber: 0,
```

Reply:
I forgot to remove these parameters from the return function. They're removed now.